### PR TITLE
Change order of steps in nominal workflows to be more consistent

### PR DESCRIPTION
**Pull request by tskisner**
Proposed changes to the order of steps in `toast_so_sim` and `toast_so_map`.

### CODE REVIEW

File: sotodlib/toast/scripts/so_sim.py [hunk 1]
```diff
@@ -70,15 +71,21 @@ def simulate_data(job, otherargs, runargs, comm):
             toast.ops.Reset(detdata=[defaults.det_data])
         # Append a weather model
         wrk.append_weather_model(job, otherargs, runargs, data)
-
+        # Before running simulations based on real data, we need
+        # a starting noise estimate
+        wrk.diff_noise_estimation(job, otherargs, runargs, data)
```
Comment (keskitalo) [inline sotodlib/toast/scripts/so_sim.py:76]: This needs to happen before zeroing the loaded data.

**Review by keskitalo — APPROVED** [review]
Only one issue about the order of operations.

File: sotodlib/toast/scripts/so_sim.py (hunk 1 above)
Comment (tskisner) [reply sotodlib/toast/scripts/so_sim.py]: Done, and the load-or-simulate logic is now extracted as we discussed offline.

**Comment by tskisner** [comment]
I think this PR can be safely merged, and then the data loading performance improvements can go in a separate PR.

**Review by keskitalo — APPROVED** [review]
Approved again.

**Pull request merged**